pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go

```diff
@@ -64,6 +64,8 @@ func (c *Config) Validate(mode config.RuntimeMode, options ...config.ValidationO
 
 	if c.MachineConfig == nil {
 		result = multierror.Append(result, errors.New("machine instructions are required"))
+
+		return result.ErrorOrNil()
```

Comment:
I understand that it fixes a panic, but I feel that we may proceed here and continue validation to show all config errors at once. Otherwise we bail out too early and skip `ClusterConfig` validation.
I guess consequent checks should properly handle `c.MachineConfig` being nil.

Reply:
looks like it will fail on everything at that point, because even node type is no longer defined, so I thought in this specific case it's better to exit early, as every other check makes little sense

Reply:
but in general, yeah, I agree, we should continue, but this is so weird that it makes little sense to continue

Reply:
ah, I see. Makes sense

Reply:
yeah, might need to continue as far as we can :) But this one looks like a fatal error